File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.java
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode res = new ListNode(-1);
        ListNode result = res;
        int total = 0;

        while (head != null) {
            if (head.val == 0 && total != 0) {
                res.next = new ListNode(total);
                res = res.next;
                total = 0;
            }
            total += head.val;
            head = head.next;
        }

        return result.next;
    }
}

File: 2181-merge-nodes-in-between-zeros/2181-merge-nodes-in-between-zeros.py
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        res = ListNode(-1)
        result = res
        total = 0

        while head:
            if head.val == 0 and total != 0:
                res.next = ListNode(total)
                res = res.next
                total = 0
            total += head.val
            head = head.next

        return result.next